memory_set/src/area.rs

```diff
@@ -140,14 +140,15 @@ impl<B: MappingBackend> MemoryArea<B> {
     /// Returns `None` if the given position is not in the memory area, or one
     /// of the parts is empty after splitting.
     pub(crate) fn split(&mut self, pos: B::Addr) -> Option<Self> {
-        let pos: usize = pos.into();
-
-        let start: usize = self.start().into();
-        let end: usize = self.end().into();
         // todo: is it a bug when `pos == end - 1`?
```

Comment:
This need to be fixed. Why is this may be a bug?

Reply:
Seems like a mistake of me, it's not a bug.

Reply:
Seems like a mistake of me, it's not a bug.